tcpclient.cpp

```diff
@@ -14,8 +14,17 @@ TcpClient::TcpClient(const QString& host, quint32 port)
     QObject::connect(m_socket.get(), &QTcpSocket::connected, this, &TcpClient::connected);
     QObject::connect(m_socket.get(), &QTcpSocket::disconnected, this, &TcpClient::disconnected);
     QObject::connect(m_socket.get(), &QTcpSocket::readyRead, this, [this]() {
-        auto msg = m_socket->readAll();
-        emit messageReceived(msg);
+        m_buffer.append(m_socket->readAll());
```

Comment:
info: for me it was hard to deduce how this code works in every possible scenario